composer/core/engine.py

```diff
@@ -1,36 +1,49 @@
 # Copyright 2021 MosaicML. All Rights Reserved.
 
-"""[summary]
-composer.Engine
-===============
+"""Engine is a coordinator for running algorithms and resolving ordering conflicts among them for composition.
 
 .. currentmodule:: composer
 
-The order of algorithms can matter significantly during composition. For example, the Selective Backprop algorithm runs during ``AFTER_DATALOADER`` event, and must run before any data augmentations. The :class:`Engine` runs these re-ordering passes.
+The order in which algorithms are run matters significantly during composition. For example,
+:class:`~.SelectiveBackprop` algorithm runs on the :attr:`~.Event.AFTER_DATALOADER` event and must run before any data
+augmentations.  :class:`~.engine.Engine` runs re-ordering passes to resolve such ordering issues or conflicts.
 
 .. note::
 
-    The design of the :class:`Engine` will be changed in future releases to accommodate more complexity as we investigation the composition of algorithms.
+    * An instance of :class:`~.engine.Engine` is automatically constructed by the :class:`~.trainer.Trainer`
+      constructor. A user need not instantiate :class:`~.engine.Engine` class.
+
+    * The design of :class:`~.engine.Engine` is subject to change in future releases to accommodate more complexity as
+      we investigate composition of algorithms.
 
 
 Currently, the following passes are registered:
 
 * **LIFO order for events**
 
-   For events that follow the ``after_*`` and ``before_*`` pattern, the ordering of algorithms is reversed for the ``after_*`` events. For example, algorithms will run in a ``ABCD -> DCBA`` ordering before and after say, the loss computation.
+   For the events that follow the ``before_*`` (e.g., :attr:`~.Event.BEFORE_LOSS`) and ``after_*`` (e.g.,
+   :attr:`~.Event.AFTER_LOSS`) pattern, the ordering of algorithms is reversed for the ``after_*`` events. For example,
+   four given algorithms ``A``, ``B``, ``C`` and ``D`` will run in ``ABCD`` ordering on the ``before_*`` event while
+   ``DCBA`` ordering on the ``after_*`` event.
 
-   This allows algorithms to "clean up" their changes. e.g. Label smoothing will smooth the labels upon entry to the loss, and then restore the original unsmoothed labels upon exit.
+   This allows algorithms to "clean up" their changes. For example, :class:`~.LabelSmoothing` will smooth the labels
+   upon on :attr:`~.Event.BEFORE_LOSS` event and then restore the original unsmoothed labels on
+   :attr:`~.Event.AFTER_LOSS` event.
 
 * **Run Selective Backprop first**
 
-   Selective backprop runs after the dataloader returns the batch, and executes an extra forward pass to rank and prune the examples in the batch by loss. To ensure a clean estimate of the example, Selective backprop should run before any other data augmentations during ``AFTER_DATALOADER`` (e.g. such as MixUp).
+   :class:`~.SelectiveBackprop` runs after the dataloader returns the batch, and executes an extra forward pass to rank
```

Comment:
Same thing as above -- can you unindent by 1 space?

Reply:
fixed